connectivity/suite.go

```diff
@@ -116,11 +119,39 @@ var (
 	client2Label = map[string]string{"name": "client2"}
 )
 
+func executeTemplate(temp *string, data any) error {
```

Comment:
nit: `renderTemplate`

Reply:
style: this should return a rendered string, rather than replacing the input.

Reply:
First I also add a function returned the rendered template. In this case we need to define extra variables to be used in WithPolicy(), they can be local variables. But this presets a challenge to use it in a for loop. How to get the rendered strings out of the for loop? Only decent solution I could think of was use a map to hold return values and also use same keys to put templates in to the for loop, or this one. I was not sure which is better. If you have any suggestions I would like to hear it, or should I go with map solution.

> style: this should return a rendered string, rather than replacing the input.

Reply:
Would internal/util package be a good place to put the renderTemplate func? or private is okay.

Reply:
I have made all the suggested changes, as I said in the previous comment I use a map to keep rendered template data and supply them to appropriate functions. I am not sure this is the best way to handle it but I could not find anything better.